src/main/java/util/RefreshTimer.java

```diff
@@ -0,0 +1,101 @@
+package util;
+
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+/**
+ * Facilitates the countdown for periodic refreshes.
+ */
+public class RefreshTimer extends TickingTimer{
```

Comment:
@dariusf previously suggests this. The name ApiQuotaManager makes the class sound like it is managing something but its just like an Utility class that only provides static methods? Or I should pass in the timer instance into class and manage it from inside?